amethyst_assets/src/loader.rs

```diff
@@ -219,85 +233,99 @@ impl Loader {
     }
 
     fn context<A>(&self) -> &Arc<A::Context>
-        where A: Asset + 'static
+    where
+        A: Asset + 'static,
     {
-        let context = self.contexts
-            .get(&TypeId::of::<A>())
-            .expect("Assets need to be registered with `Loader::register`.");
+        let context = self.contexts.get(&TypeId::of::<A>()).expect(
+            "Assets need to be registered with `Loader::register`.",
+        );
 
         context.downcast_ref().unwrap()
     }
 
     fn store<S>(&self, store: &S) -> Arc<Box<AnyStore>>
-        where S: Eq + Hash + ?Sized,
-              String: Borrow<S>
+    where
+        S: Eq + Hash + ?Sized,
+        String: Borrow<S>,
     {
         self.stores
             .get(&store)
-            .expect("No such store. Maybe you forgot to add it with `Loader::add_store`?")
+            .expect(
+                "No such store. Maybe you forgot to add it with `Loader::add_store`?",
+            )
             .clone()
     }
 }
 
 /// Loads an asset with a given context, format, specifier and storage right now.
-pub fn load_asset<A, F, N, S>(context: &A::Context,
-                              format: &F,
-                              name: N,
-                              storage: &S)
-                              -> Result<A, AssetError<A::Error, F::Error, S::Error>>
-    where A: Asset,
-          F: Format<Data = A::Data>,
-          N: Into<String>,
-          S: Store
+pub fn load_asset<A, F, N, S>(
+    context: &A::Context,
+    format: &F,
+    name: N,
+    storage: &S,
+) -> Result<A, AssetError<A::Error, F::Error, S::Error>>
+where
+    A: Asset,
+    F: Format<Data = A::Data>,
+    N: Into<String>,
+    S: Store,
 {
     let name = name.into();
 
     let store_id = storage.store_id();
-    let spec = AssetSpec::new(name.clone(), F::extension(), store_id);
+    let spec = AssetSpec::new(name.clone(), F::extensions(), store_id);
 
     A::retrieve(context, &spec)
```

Comment:
We can think about throwing an exception if multiple files with some name and different extensions exists.